src/main/java/org/prebid/server/floors/PriceFloorFetcher.java

```diff
@@ -0,0 +1,316 @@
+
+        return fetchContext.getRules();
+    }
+
+    private long resolveCacheTtl(ResponseCacheInfo cacheInfo, AccountPriceFloorsFetchConfig fetchConfig) {
+        final Long responseTtl = cacheInfo.getCacheTtl();
+        if (responseTtl != null) {
+            return responseTtl;
+        }
+
+        return ObjectUtil.getIfNotNull(fetchConfig, AccountPriceFloorsFetchConfig::getMaxAgeSec);
+    }
+
+    private Long createMaxAgeTimer(String accountId, long cacheTtl) {
+        final Long previousTimerId =
+                ObjectUtil.getIfNotNull(fetchedData.get(accountId), AccountFetchContext::getMaxAgeTimerId);
+
+        if (previousTimerId != null) {
+            vertx.cancelTimer(previousTimerId);
+        }
+
+        return vertx.setTimer(TimeUnit.SECONDS.toMillis(cacheTtl), id -> fetchedData.remove(accountId));
+    }
+
+    private Future<PriceFloorRules> recoverFromFailedFetching(Throwable throwable, String accountId) {
+        metrics.updatePriceFloorFetchMetric(MetricName.failure);
+
+        if (throwable instanceof TimeoutException || throwable instanceof ConnectTimeoutException) {
+            samplingLogger.warn(
+                    String.format("Fetch price floor request timeout for account %s exceeded.", accountId), 0.01);
+        } else if (throwable instanceof PreBidException) {
+            samplingLogger.warn(
+                    String.format("Failed to fetch price floor from provider for account = %s with a reason : %s ",
+                            accountId, throwable.getMessage()), 0.01);
+        }
+        fetchInProgress.remove(accountId);
+
+        return Future.succeededFuture();
+    }
+
+    private PriceFloorRules createPeriodicTimerForRulesFetch(PriceFloorRules priceFloorRules,
+                                                             AccountPriceFloorsFetchConfig fetchConfig,
+                                                             String accountId) {
+        final long periodicTime = ObjectUtil.getIfNotNull(fetchConfig, AccountPriceFloorsFetchConfig::getPeriodSec);
+        vertx.setTimer(TimeUnit.SECONDS.toMillis(periodicTime), id -> periodicFetch(accountId));
+
+        return priceFloorRules;
+    }
+
+    private void periodicFetch(String accountId) {
+
+        accountById(accountId).compose(this::fetchPriceFloorRules);
+    }
+
+    private Future<Account> accountById(String accountId) {
+        return StringUtils.isBlank(accountId)
+                ? Future.succeededFuture()
+                : applicationSettings
+                .getAccountById(accountId, timeoutFactory.create(TimeUnit.SECONDS.toMillis(ACCOUNT_FETCH_TIMEOUT)))
+                .recover(ignored -> Future.succeededFuture());
+    }
+
+    @Value(staticConstructor = "of")
+    private static class AccountFetchContext {
+
+        PriceFloorRules rules;
+
+        Long maxAgeTimerId;
+    }
+
+    @Value(staticConstructor = "of")
+    private static class ResponseCacheInfo {
```

Comment:
Can it be renamed to `PriceFloorResponse` ? i don't see the reason to have `Cache` in name.

Reply:
We have cached data + cache time to live, `Response`  in name means only body of HttpResponse or full http response, so seems cache info better describes data inside